src/libsyntax/parse/parser.rs

```diff
@@ -1661,6 +1661,8 @@ impl<'a> Parser<'a> {
 
     /// matches '-' lit | lit
     pub fn parse_pat_literal_maybe_minus(&mut self) -> PResult<'a, P<Expr>> {
+        maybe_whole_expr!(self);
```

Comment:
Should the check be in `ast_validation` or right here in `parse_pat_literal_maybe_minus`?